#! /bin/sh -e
## 10_a_capella.dpatch by  <kyle@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix UTF16 parsing bug

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -u libid3tag-0.15.1b.old/utf16.c libid3tag-0.15.1b/utf16.c
--- libid3tag-0.15.1b.old/utf16.c
+++ libid3tag-0.15.1b/utf16.c	
@@ -250,6 +250,8 @@
   id3_ucs4_t *ucs4;
 
   end = *ptr + (length & ~1);
+  if (end == *ptr)
+    return 0;
 
   utf16 = malloc((length / 2 + 1) * sizeof(*utf16));
   if (utf16 == 0)
